### ISSUE TITLE
Add availability to Fail Test Suite

### ISSUE THREAD

**Issue by myjniak** [robotframework/robotframework#3493]
The source of our problem is that our test cases cannot be independent.
We run multiple Test Suites per robot execution where each Suite IS independent.
We would like to have a feature like:
1. New KW:  Fail Test Suite  msg=xxx
2. New arg:  Fail  msg=xxx  level=suite
or:
3. New arg:  Fatal Error  msg=xxx   level=suite

Because right now, we only have the following solution:
```
*** Settings ***
Test Setup    Run Keyword If  "${SUITE_STATUS}"=="FAIL"  Fail  blah blah
*** Variables ***
${SUITE_STATUS}  PASS

*** Test Cases ***
Test 1
    Log  1
Test 2
    Log  2
    Run Suite-Critical Action
Test 3
    Log  3

*** Keywords ***
Fail Test Suite
    Set Suite Variable  ${SUITE_STATUS}  FAIL
    Fail  blah blah

Run Suite-Critical Action
    Suite Critical Action
    [Teardown]  Run Keyword If  "${KEYWORD_STATUS}"=="FAIL"  Fail Test Suite
```

The thing is that we have multiple Test Suites where such logic should be copy-pasted, and some Test Setups require some other actions.

We also tried to override start_test():
```
    def start_test(self, name, attributes):
        if suite_failed():
            raise AssertionError("Suite-critical error occured, skipping test")
```
but it results in an Log  ERROR from ListenerMethod class.

Is there any other way we could do it nicely? As an output we would like to have a KW "Fail Test Suite" which doesn't require any repetitive actions/variable definitions per Suite.

**Comment by pekkaklarck** [comment]
Do I get it right that you want to stop executing a suite so that tests in it fail but continue from the next one? If yes, you are right that it's not directly supported. Related features that we have include:

- Failures in suite setup stop executing that suite. All tests in it will be marked failed and execution continues from the next suite.
- Automatically set `${PREV TEST STATUS}` variable can be used to check did the previous test pass or not.
- `Fatal Error` keyword and custom exceptions that stop the whole execution.

Having an easy way to fail remaining tests in a suite without stopping the whole execution would be a handy feature. Unfortunately it is somewhat big task and there are lot of other issues with higher priority and thus it's unlikely that I'll have time to think about it more in the foreseeable future. If someone wants to get it soon, they should consider looking at implementing it and providing a PR (I can help and review) or sponsoring the development via [Robot Framework Foundation](https://robotframework.org/foundation/) or otherwise.

**Comment by myjniak** [comment]
Thank you for your fast response,
We also may have problems with time resources, but maybe we'll try to implement this to the Robot code by ourselves,

Do you have any technical guidelines / preferred way of making this keyword happen? 
What makes you think it's a big feature?

**Comment by pekkaklarck** [comment]
It's not likely to be a huge task but there's quite a bit of work with implementation, tests and docs. The first task is designing how this feature should work. Do you have ideas?

**Comment by pekkaklarck** [comment]
The bigger question is, what would be the way to initiate this? Keyword in BuiltIn? Custom exception? Both? Something else?

**Comment by myjniak** [comment]
Honestly I was only thinking about new keyword, or additional arg like 'scope' to Fail keyword.
Fail is not a custom exception right? It just raises AssertionError. I would like to keep the scheme.

**Comment by pekkaklarck** [comment]
Yes, `Fail` just raises a regular `AssertionError`. This is, however, much closer to `Fatal Error` that uses a custom exception  or actually an exception with a custom attribute. If this support is implemented, I find it important that keywords can directly initiate it. Always needing to use a specific keyword would be inconvenient.

**Comment by myjniak** [comment]
You're saying to create somthing like error.ROBOT_END_SUITE_ON_FAILURE accordingly to this?:
```
    def fatal_error(self, msg=None):
        error = AssertionError(msg) if msg else AssertionError()
        error.ROBOT_EXIT_ON_FAILURE = True
        raise error
```

**Comment by pekkaklarck** [comment]
Yes. For consistency reasons the attribute could be named `ROBOT_EXIT_SUITE_ON_FAILURE`.

Probably would be a good idea to expose an exception with that attribute set as well to ease the usage. Should do the same with existing fatal and continuable exceptions as well but that requires a separate issue.

**Comment by vfuksa-cen89549** [comment]
I'd like to support the request for this feature. My use case is however different: I want all tests in a suite to fail if the test setup of a test from this suite fails. The reason is that all tests in a suite share the same setup and thus when a setup of a test fails, all tests will probably fail. 

Having a workaround in form of a suite-level variable stating that the tests in the suite should be skipped would require:
- explicitely creating keywords used as test setups (instead of using Run Keywords ... AND ...), so that they could have the teardowns, where the variable can be set when the keyword fails,
- checking the variable as the first thing of each test setup
- explicitely creating keywords used as test teardowns, so that tag robot:recursive-stop-on-failure could be set to them and the state of the variable could be checked as the first thing (however, in other cases I wish all the KWs in a teardown are executed even if one of them fails)

Listeners seem to help just a bit with this annoying procedure and code repetitions. 

I am for creating a KW or another mechanism enabling Robot to skip (or immediately fail) an execution of (remaining tests of) a suite under a condition.
